src/gui/qgsattributeform.cpp

```diff
@@ -1789,6 +1814,14 @@ void QgsAttributeForm::init()
                 mLabelDataDefinedProperties[ label ] = property;
               }
             }
+            if ( mLayer->editFormConfig().dataDefinedFieldProperties( fieldName ).hasProperty( QgsEditFormConfig::DataDefinedProperty::Editable ) )
+            {
+              const QgsProperty property { mLayer->editFormConfig().dataDefinedFieldProperties( fieldName ).property( QgsEditFormConfig::DataDefinedProperty::Editable ) };
+              if ( property.isActive() && ! property.expressionString().isEmpty() )
```

Comment:
If the property is a field based property, then the expressionString will always be empty and this logic won't work.

```suggestion
              if ( property.isActive() )
```